mesh/mesh.c

```diff
@@ -445,7 +452,7 @@ static bool prov_complete_cb(void *user_data, uint8_t status,
 		return false;
 
 	owner = join_pending->sender;
-	path = join_pending->app_path;
+	path = node_get_app_path(join_pending->node);
```

Comment:
This is worth a separate commit, I think - it fixes a genuine bug in the daemon.

Reply:
Also, is it possible to remove `app_path` from `join_pending` altogether?